core/chaincode/handler.go

```diff
@@ -1077,6 +1079,36 @@ func (h *Handler) HandleDelState(msg *pb.ChaincodeMessage, txContext *Transactio
 	return &pb.ChaincodeMessage{Type: pb.ChaincodeMessage_RESPONSE, Txid: msg.Txid, ChannelId: msg.ChannelId}, nil
 }
 
+func (h *Handler) HandlePurgePrivateData(msg *pb.ChaincodeMessage, txContext *TransactionContext) (*pb.ChaincodeMessage, error) {
+	delState := &pb.DelState{}
+	err := proto.Unmarshal(msg.Payload, delState)
+	if err != nil {
+		return nil, errors.Wrap(err, "unmarshal failed")
+	}
+
+	namespaceID := txContext.NamespaceID
+	collection := delState.Collection
+	if isCollectionSet(collection) {
+		if txContext.IsInitTransaction {
+			return nil, errors.New("private data APIs are not allowed in chaincode Init()")
+		}
+		if err := errorIfCreatorHasNoWritePermission(namespaceID, collection, txContext); err != nil {
+			return nil, err
+		}
+		err = nil
+		// TODO  will call code from issue #3205
+		// txContext.TXSimulator.PurgePrivateData(namespaceID, collection, delState.Key)
+	} else {
+		return nil, errors.New("only applicable to for private data")
```

Comment:
In idiomatic Go we try to remove as many if/else branches as possible.
In this case you can first return the error if isCollectionSet() returns false, and THEN process the above logic in the main path.  Said another way, since this function is only intended for private data, better to validate that condition first and then there is no need for the "if isCollection()" statement.
Another reason this will flow better is that the following err check will be immediately after the corresponding logic, rather than having this else statement between the logic and the final error check.

Reply:
I've re-ordered the functions as suggested....